### PR TITLE
Convert MNIST to Swift Manager

### PR DESCRIPTION
**Pull request by xiejw**
1. Rename the mnist.swift to main.swift so swiftpm can find it.
2. Improve the readFile method to search folders. This is required as we are changing the folder to run it. 
3. Give a better error message when file cannot be found.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// swift-tools-version:4.2
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "Swift-Models",
```
Comment (rxwei) [inline Package.swift]:
```suggestion
    name: "TensorFlowModels",
```

File: MNIST/main.swift [hunk 2]
```diff
@@ -12,15 +12,27 @@
...
 
 /// Reads a file into an array of bytes.
 func readFile(_ filename: String) -> [UInt8] {
-    let d = Python.open(filename, "rb").read()
-    return Array(numpy: np.frombuffer(d, dtype: np.uint8))!
+    let possibleFolders = [".", "MNIST"]
+    for folder in possibleFolders {
+        let parent = URL(fileURLWithPath: folder)
+        let filePath = parent.appendingPathComponent(filename).path
+
```
Comment (rxwei) [inline MNIST/main.swift]: I think this extra line is not improving clarity and could be removed.

File: MNIST/main.swift [hunk 3]
```diff
@@ -12,15 +12,27 @@
...
+        guard FileManager.default.fileExists(atPath: filePath) else {
+            continue
+        }
+
```
Comment (rxwei) [inline MNIST/main.swift]: This too.

File: MNIST/main.swift [hunk 4]
```diff
@@ -12,15 +12,27 @@
...
+        let d = Python.open(filePath, "rb").read()
+        return Array(numpy: np.frombuffer(d, dtype: np.uint8))!
+    }
+    fatalError("Failed to find file with name \(filename) in following folders: \(possibleFolders)")
```
Comment (rxwei) [inline MNIST/main.swift]: Fit within 100 columns.

File: Package.swift [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    targets: [
```
Comment (rxwei) [inline Package.swift]: Maybe explicitly define `products:` as well?

File: MNIST/main.swift (hunk 2 above)
Comment (xiejw) [reply MNIST/main.swift]: Deleted.

File: MNIST/main.swift (hunk 3 above)
Comment (xiejw) [reply MNIST/main.swift]: deleted.

File: MNIST/main.swift (hunk 4 above)
Comment (xiejw) [reply MNIST/main.swift]: This line is exactly 100 columns.

File: Package.swift (hunk 5 above)
Comment (xiejw) [reply Package.swift]: Done.

File: MNIST/main.swift [hunk 6]
```diff
@@ -12,15 +12,25 @@
...
+    for folder in possibleFolders {
+        let parent = URL(fileURLWithPath: folder)
+        let filePath = parent.appendingPathComponent(filename).path
+        guard FileManager.default.fileExists(atPath: filePath) else {
+            continue
+        }
+        let d = Python.open(filePath, "rb").read()
+        return Array(numpy: np.frombuffer(d, dtype: np.uint8))!
+    }
+    fatalError("Failed to find file with name \(filename) in following folders: \(possibleFolders)")
```
Comment (rxwei) [inline MNIST/main.swift]:
```suggestion
    preconditionFailure("Failed to find file with name \(filename) in the following folders: \(possibleFolders)")
```

Now you'll need to break the line :)

Comment (xiejw) [reply MNIST/main.swift]: As discussed, we want an error message so we will use the old one.

Comment (rxwei) [reply MNIST/main.swift]: I did also suggest a grammar-related change though.

**Pull request merged**